src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs

```diff
@@ -0,0 +1,337 @@
+                {
+                    string organization = pipeline.ReleasePipeline.Organization;
+                    string project = pipeline.ReleasePipeline.Project;
+                    int pipelineId = pipeline.ReleasePipeline.PipelineIdentifier;
+
+                    AzureDevOpsReleaseDefinition pipeDef = await azdoClient.GetReleaseDefinitionAsync(organization, project, pipelineId);
+                    pipeDef = await azdoClient.RemoveAllArtifactSourcesAsync(organization, project, pipeDef);
+
+                    pipeDef = await azdoClient.AddArtifactSourceAsync(organization, project, pipeDef, azdoBuild);
+
+                    int releaseId = await azdoClient.StartNewReleaseAsync(organization, project, pipeDef, build.Id);
+
+                    var item = new ReleasePipelineStatusItem(releaseId, channelId, organization, project);
+                    releaseList.Add(item);
+
+                }
+                catch (Exception e)
+                {
+                    Logger.LogError($"Some problem happened while starting publishing pipeline " +
+                        $"{pipeline.ReleasePipeline.PipelineIdentifier} for build " +
+                        $"{build.AzureDevOpsBuildId}: {e.Message}", e);
+                    throw;
+                }
+            }
+
+            if (releaseList.Count > 0)
+            {
+                using (ITransaction tx = StateManager.CreateTransaction())
+                {
+                    var runningPipelinesForBuild = await runningPipelines.TryGetValueAsync(tx, buildId);
+                    if (runningPipelinesForBuild.HasValue)
+                    {
+                        // Some channel already triggered release pipelines for this build. Need to update with the releases for the new channel.
+                        releaseList.AddRange(runningPipelinesForBuild.Value);
+                        await runningPipelines.AddOrUpdateAsync(tx, buildId, releaseList, (key, oldValue) => releaseList);
+                    }
+                    else
+                    {
+                        await runningPipelines.AddAsync(tx, buildId, releaseList);
+                    }
+                    await tx.CommitAsync();
+                }
+            }
+            await ProcessFinishedReleasesAsync(cancellationToken);
+        }
+
+        /// <summary>
+        /// Check for finished releases and add the corresponding build channel association to the DB.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        // Check every 10 minutes for any finished release Pipelines
+        [CronSchedule("0 0/10 * 1/1 * ? *", TimeZones.UTC)]
+        public async Task ProcessFinishedReleasesAsync(CancellationToken cancellationToken)
+        {
+            var runningPipelines =
+                        await StateManager.GetOrAddAsync<IReliableDictionary<int, IList<ReleasePipelineStatusItem>>>(RunningPipelineDictionaryName);
```

Comment:
This indentation is weird: why so far indented?  It didn't make the line much smaller either.  Unfortunately, most of it is the huge generic, so I don't have a good line breaking plan that looks good. :-)

Reply:
Weird indentation was probably from refactoring something at some point and I just didn't realize.